src/actions/secrets.js

```diff
@@ -56,10 +58,72 @@ export function fetchSecrets({ namespace } = {}) {
 export function deleteSecret(secrets) {
   return async dispatch => {
     dispatch({ type: 'SECRET_DELETE_REQUEST' });
+    // This is where we first handle the unpatching (complicated section), and would benefit
+    // from error handling and additional notifications on success/error
+    const namespacesToSecretsMap = new Map();
+    secrets.forEach(secret => {
+      let foundSecretInfo = namespacesToSecretsMap.get(secret.namespace);
+      if (foundSecretInfo) {
+        foundSecretInfo.push(secret.name);
+      } else {
+        foundSecretInfo = [secret.name];
+      }
+      namespacesToSecretsMap.set(secret.namespace, foundSecretInfo);
+    });
+
+    /* Now we know the secrets for each namespace, iterate and determine which secrets 
+    should be removed from the SA. With the list of known secrets that stay, 
+    we finally do a replace type PATCH on the entire service account at once. 
+    This is used to ensure all data is still correct regardless of things like indexes changing 
+    which is the only way to remove secrets otherwise using the patch API */
+
+    // For each namespace there are secrets in
+    namespacesToSecretsMap.forEach(async function(value, namespace) {
+      // Get all the service accounts in the namespace
+      const serviceAccounts = await getServiceAccounts({
+        namespace
+      });
+
+      // For each service account found...
+      serviceAccounts.forEach(async serviceAccount => {
+        let secretRemoved = false;
+        const remainingSecrets = [];
+        const allSecrets = serviceAccount.secrets;
+        // For each secret found
+        for (let x = 0; x < allSecrets.length; x += 1) {
+          const secret = allSecrets[x].name;
+          let found = false;
+          // For each secret we want to delete
+          for (let z = 0; z < value.length; z += 1) {
+            const item = value[z];
+            // Name matches?
+            if (item === secret) {
+              found = true;
+              secretRemoved = true;
+            }
+          }
+          // Didn't find one to delete, so update listing of ones to keep
+          if (!found) {
+            const itemAsJson = { name: secret };
+            remainingSecrets.push(itemAsJson);
+          }
+        }
+        // One's been removed so we'll need to now replace the
+        // ServiceAccount's secrets field with the ones to keep
+        if (secretRemoved) {
+          replaceSecretFromServiceAccount(
```

Comment:
This method name is just a bit clunky - `patchSecretElementOnServiceAccount` or `updateServiceAccountSecrets` perhaps? Again it's only a minor nit.